Tests/KeyboardToolbarTests.swift

```diff
@@ -0,0 +1,134 @@
+        keyboardToolbar.addButton(barButton)
+        
+        XCTAssertEqual(keyboardToolbar.items?.first, barButton)
+    }
+    
+    func test_KeyboardToolbar_AddBackButtonWithTitle() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addBackButton(title: "Title")
+
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_AddBackButtonWithImage() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addBackButton(image: UIImage())
+        
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_AddNextButtonWithTitle() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addNextButton(title: "Title")
+        
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_AddNextButtonWithImage() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addNextButton(image: UIImage())
+        
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_AddDoneButtonWithTitle() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addDoneButton(title: "Title")
+        
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_AddDoneButtonWithImage() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addDoneButton(image: UIImage())
+        
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_AddSystemDoneButton() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addSystemDoneButton()
+        
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_AddFlexibleSpace() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.addFlexibleSpace()
+        
+        XCTAssertEqual(keyboardToolbar.items?.count, 1)
+    }
+    
+    func test_KeyboardToolbar_BackButtonTapped() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.keyboardAccessoryDelegate = self
+        keyboardToolbar.addBackButton(title: "Title")
+        keyboardToolbar.backButtonTapped(keyboardToolbar.items!.first!)
+    }
+    
+    func test_KeyboardToolbar_NextButtonTapped() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.keyboardAccessoryDelegate = self
+        keyboardToolbar.addNextButton(title: "Title")
+        keyboardToolbar.nextButtonTapped(keyboardToolbar.items!.first!)
+    }
+    
+    func test_KeyboardToolbar_DoneButtonTapped() {
+        let keyboardToolbar = KeyboardToolbar()
+        keyboardToolbar.keyboardAccessoryDelegate = self
+        keyboardToolbar.addDoneButton(title: "Title")
+        keyboardToolbar.doneButtonTapped(keyboardToolbar.items!.first!)
+    }
+}
+
+extension KeyboardToolbarTests: KeyboardAccessoryDelegate {
```

Comment:
Similar comment to above. I think using a mock would make more sense for testing these things.